app/presenters/vm_presenter.rb

```diff
@@ -0,0 +1,12 @@
+class VmPresenter
+  def initialize(controller, session)
+    @controller = controller
```

Comment:
I missed removing the passed in controller when refactoring the original code, thanks for catching that.  The original code also made use of a ```Vm``` but that turned out to be dead code and was removed.  Now in hindsight I see that I can pass the session value directly to the function as well.  I'll make these changes momentarily.